### PR TITLE
test: expand integration tests for SDK client

### LINKED ISSUE #2747: Expand integration tests for SDK client
### Problem

Currently there is limited coverage on the SDK client in the `sdkClient.spec.ts` file.

### Solution

Add integration tests covering `createFile()` (with `callData.length` > `fileAppendChunkSize` and without), `deleteFile()`, `executeQuery`, `executeTransaction`, `executeGetTransactionRecord ` and verify that `hbarLimiter.addExpense()` has been called with the transaction cost (e.g., with a tolerance of 0.1% for delta)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

This PR adds integration tests covering:
- `createFile()` (with `callData.length > fileAppendChunkSize` and `callData.length <= fileAppendChunkSize`)
- `deleteFile()`
- `executeQuery()`
- `executeTransaction()`
- `executeGetTransactionRecord()`.  

The tests verify that:
1. `hbarLimiter.shouldLimit()` has been called the appropriate amount of times before executing the transaction
2. `hbarLimiter.addExpense()` has been called the appropriate amount of times with the cost of each transaction

**Related issue(s)**:

Fixes #2747 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/clients/sdkClient.ts [hunk 1]
```diff
@@ -817,12 +834,18 @@ export class SDKClient {
...
    */
-  public deleteFile = async (fileId: FileId, requestId?: string, callerName?: string, interactingEntity?: string) => {
+  async deleteFile(fileId: FileId, requestId: string, callerName: string, interactingEntity: string): Promise<void> {
     // format request ID msg
     const currentDateNow = Date.now();
     const requestIdPrefix = formatRequestIdMessage(requestId);
 
     try {
+      // TODO: Limiting was missing here, not sure if it was intentional
+      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/sdkClient.ts:844]: Was intentional.
We always want to clean up fies so we don't want the rate limiter to block us in this case.

File: packages/relay/tests/lib/sdkClient.spec.ts [hunk 2]
```diff
@@ -69,7 +74,7 @@ describe('SdkClient', async function () {
 
     client = client.setOperator(
       AccountId.fromString(process.env.OPERATOR_ID_MAIN!),
-      PrivateKey.fromString(process.env.OPERATOR_KEY_MAIN!),
+      PrivateKey.fromStringDer(process.env.OPERATOR_KEY_MAIN!),
```
Comment (Nana-EC) [inline packages/relay/tests/lib/sdkClient.spec.ts:77]: Q: why the change as part of this PR?
Does this force DER only?
I believe we wanted to move to a place that supported DER or hex for EVM tools.
If not related this should be reverted and tackled in another PR

File: packages/relay/tests/lib/sdkClient.spec.ts [hunk 3]
```diff
@@ -2045,31 +2067,203 @@ describe('SdkClient', async function () {
       124, 18, 190, 114, 79, 189, 99, 27, 157, 117, 227, 107, 74, 255, 111, 74, 166, 7, 78, 25, 35, 131, 85, 47, 52,
       120, 20,
     ]);
+    const accountId = AccountId.fromString('0.0.1234');
```
Comment (Nana-EC) [inline packages/relay/tests/lib/sdkClient.spec.ts:2070]: Should make accountId configurable in case a test runner needs to change the account. The current value is the default

File: packages/relay/tests/lib/sdkClient.spec.ts [hunk 4]
```diff
@@ -2045,31 +2067,203 @@ describe('SdkClient', async function () {
...
+
+      await sdkClient.executeGetTransactionRecord(
+        transactionResponse,
+        callerName,
+        interactingEntity,
+        new EthereumTransaction().constructor.name,
+      );
+
+      hbarLimitMock.verify();
+    });
```
Comment (Nana-EC) [inline packages/relay/tests/lib/sdkClient.spec.ts:2267]: This is really good.
Is it possible to add one test to ensure all the above checks happen in a flow for all the transactions and queries for say a transaction than needs 2 file appends?

File: packages/relay/tests/lib/sdkClient.spec.ts (hunk 2 above)
Comment (victor-yanev) [reply packages/relay/tests/lib/sdkClient.spec.ts:77]: The `fromString` method was getting flagged for being deprecated by eslint, and since we use DER keys everywhere in our `.env` files I changed this to `fromStringDer`. Please let me know if I'm missing anything and I should revert this.

File: packages/relay/tests/lib/sdkClient.spec.ts (hunk 3 above)
Comment (victor-yanev) [reply packages/relay/tests/lib/sdkClient.spec.ts:2070]: Those are only mocked values, we don't do any real requests here

File: packages/relay/tests/lib/sdkClient.spec.ts (hunk 4 above)
Comment (victor-yanev) [reply packages/relay/tests/lib/sdkClient.spec.ts:2323]: Done:
- I added a new test `should rate limit before creating file and add expenses to limiter for large transaction data` which executes an `EthereumTransaction` with `callData` which needs 13 file append chunks - this is the most detailed test of all, also covers expenses for `FileContentsQuery` to retrieve the `FeeSchedules`
- I also updated the test `should execute FileCreateTransaction with callData.length > fileAppendChunkSize and add expenses to limiter` to use `callData.length := FILE_APPEND_CHUNK_SIZE * 2 + 1`

File: packages/relay/tests/lib/sdkClient.spec.ts (hunk 2 above)
Comment (Nana-EC) [reply packages/relay/tests/lib/sdkClient.spec.ts:95]: Gotcha. 
It's only affecting the tests path in this case so it should be fine.
Carry on.

**Review by Nana-EC — APPROVED** [review]

File: packages/relay/tests/lib/sdkClient.spec.ts (hunk 2 above)
Comment (quiet-node) [reply packages/relay/tests/lib/sdkClient.spec.ts:94]: Eric wrote a method to create private key based on format here https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/utils.ts#L44-L56. Can this be employed here to make it more dynamic?

Comment (victor-yanev) [reply packages/relay/tests/lib/sdkClient.spec.ts]: Replaced it with `createPrivateKeyBasedOnFormat `, thanks!

**Review by quiet-node — APPROVED** [review]
LGTM. 1 small QQ

File: packages/relay/tests/lib/sdkClient.spec.ts [hunk 5]
```diff
@@ -150,24 +185,11 @@ describe('SdkClient', async function () {
...
-                    servicedata: undefined,
-                  },
-                ],
-              },
-            ],
-          },
-        };
+        return Promise.resolve(feeSchedules);
       });
+      // @ts-ignore
```
[the remark this answers is not in the document]
Comment (victor-yanev) [reply packages/relay/tests/lib/sdkClient.spec.ts:189]: Cause it was producing typescript warnings

**Review by ebadiere — APPROVED** [review]
LG.
